lager/lenses.hpp

```diff
@@ -123,51 +129,169 @@ auto attr(Member member)
     });
 }
 
+/*!
+ * `Key -> Lens<{X}, [X]>`
+ */
 template <typename Key>
-auto at(Key key)
-{
-    return zug::comp([=](auto&& f) {
-        return [f, &key](auto&& p) {
-            return f([&] {
+auto at(Key key) {
+    return zug::comp([key](auto&& f) {
+        return [f = LAGER_FWD(f), &key](auto&& whole) {
+            using Part = std::optional<std::decay_t<decltype(whole.at(key))>>;
+
+            return f([&]() -> Part {
                 try {
-                    return std::forward<decltype(p)>(p).at(key);
-                } catch (std::out_of_range const&) {
-                    return std::decay_t<decltype(p.at(key))>{};
+                    return std::forward<decltype(whole)>(whole).at(key);
+                } catch (std::out_of_range const&) { return std::nullopt; }
+            }())([&](Part part) {
+                auto r = std::forward<decltype(whole)>(whole);
+                if (part.has_value()) {
+                    try {
+                        r.at(key) = std::move(part).value();
+                    } catch (std::out_of_range const&) {}
                 }
-            }())([&](auto&& x) {
-                auto r = std::forward<decltype(p)>(p);
-                try {
-                    r.at(key) = std::forward<decltype(x)>(x);
-                } catch (std::out_of_range const&) {}
                 return r;
             });
         };
     });
 }
 
+/*!
+ * `Key -> Lens<{X}, [X]>`
+ */
 template <typename Key>
-auto at_i(Key key)
-{
-    return zug::comp([=](auto&& f) {
-        return [f, &key](auto&& p) {
-            return f([&] {
+auto at_i(Key key) {
```

Comment:
Also... I have been doubting a lot about this function, maybe it should be `at_immer`.

Reply:
maybe  `lager::lens::immer::at()`?
this way we could also have lenses for other immer containers like `lens::immer::unbox()`